openpgp/eax/eax.go

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) 2019 ProtonTech AG
+// [1]  https://web.cs.ucdavis.edu/~rogaway/papers/eax.pdf
+
+package eax
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+)
+
+const (
+	defaultTagSize   = 16
+	defaultNonceSize = 16
+)
+
+type eax struct {
+	block     cipher.Block // Only AES-128 supported
+	tagSize   int          // At least 12 bytes recommended
+	nonceSize int
+}
+
+// NewEAX returns an EAX instance with AES-128 and default parameters.
+// Input key length must equal 128 bits
+func NewEAX(key []byte) eax {
+	if len(key) != 16 {
+		panic(
+			`Error: EAX only supports AES-128, currently. Please supply a
+			 128-bit key`)
+	}
```

Comment:
Yes, it works for key lengths {128, 192, 256} just fine. The reason I included this was because of the [draft](https://tools.ietf.org/html/draft-ietf-openpgp-rfc4880bis-07#section-5.16.1) (5.16.1), but now I realize that there is no MUST about it. I'll drop the restriction, and panic for invalid key lengths will be transmitted from the underlying cipher.

Reply:
(And I'm modifying the TestEncryptDecryptRandVectors test to cover these three cases)